src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -80,6 +80,9 @@ public void validate(@Nullable Schema inputSchema) {
     super.validate();
     if (!containsMacro("schema")) {
       Schema outputSchema = getSchema();
+      if (timePartitioningAllowed) {
+        BigQueryUtil.validateColumnForPartition(partitionByField, outputSchema);
+      }
```

Comment:
we also need to handle the scenario where the table exists but it doesn't match the partitioning values that have been configured. For example:

1. Table is not partitioned but the user has specified a partition column
2. Table is partitioned but on a column that doesn't match the user provided column.
3. Table is requires a partition filter and the user did not provide a partition column

Reply:
Done.